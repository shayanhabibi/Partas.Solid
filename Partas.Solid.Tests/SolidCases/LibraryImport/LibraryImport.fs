module Partas.Solid.Tests.SolidCases.LibraryImport

open Partas.Solid
open Fable.Core
open Partas.Solid.Tests.Shared.LibraryImport


type [<Erase>] MyTag() =
    interface RegularNode
    [<SolidTypeComponent>]
    member props.typeDef =
        Imported()

type [<Erase>] MyOtherTag() =
    interface RegularNode
    [<SolidTypeComponent>]
    member props.typeDef =
        CustomTag()
        
type [<Erase>] ImportedTagWithProps() =
    interface RegularNode
    [<SolidTypeComponent>]
    member props.typeDef =
        Imported(class' = "Test props")

type [<Erase>] UserTagWithProps() =
    interface RegularNode
    [<SolidTypeComponent>]
    member props.typeDef =
        CustomTag(class' = "test bops")
