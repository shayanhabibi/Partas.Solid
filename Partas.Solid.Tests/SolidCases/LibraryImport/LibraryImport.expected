import { splitProps } from "solid-js";
import { FakeImport } from "FakeLibrary";
import { CustomTag } from "./Shared.fs.jsx";
export function MyTag(props) {
    const [PARTAS_LOCAL, PARTAS_OTHERS] = splitProps(props, []);
    return <FakeImport />;
}
export function MyOtherTag(props) {
    const [PARTAS_LOCAL, PARTAS_OTHERS] = splitProps(props, []);
    return <CustomTag />;
}
export function ImportedTagWithProps(props) {
    const [PARTAS_LOCAL, PARTAS_OTHERS] = splitProps(props, []);
    return <FakeImport class="Test props" />;
}
export function UserTagWithProps(props) {
    const [PARTAS_LOCAL, PARTAS_OTHERS] = splitProps(props, []);
    return <CustomTag class="test bops" />;
}